### ISSUE TITLE
How to access linear memory

### ISSUE THREAD

**Issue by robinvanemden** [wasm3/wasm3#45]
To access non-integer results from WebAssembly modules, I generally first retrieve respectively the location and the length of a result variable as previously allocated to the linear memory of the WebAssembly instance. This enables us to consequently read (or write) to that memory area. For a `Golang` example, see for instance:

https://github.com/wasmerio/go-ext-wasm#read-the-memory
```go
memory := instance.Memory.Data()
// Reads the memory.
result := memory[pointer : pointer+length]
// now you can parse "result" 
```
I presume that in `wasm3`, I would need to access `M3Memory`? I am, however, unsure how to interface with it. Could you suggest how best to proceed to obtain access to linear memory from `wasm3`?

**Comment by vshymanskyy** [comment]
Right. There's no API to access memory yet, but it can be added easily. Will be working on this soon.

**Comment by soundandform** [comment]
I just added an m3_GetMemory API call

**Comment by robinvanemden** [comment]
That is fast! Thanks, both @soundandform and @vshymanskyy. As it is almost midnight here, I will have to wait untill tomorrow morning to take the new m3_GetMemory API call for a spin. I will be able find out tomorrow anyway, but out of curiosity, is it possible to use the API call to both read from and write to linear memory? Or does the function name imply read access only?

**Comment by soundandform** [comment]
It simply returns a raw pointer to memory.  You can read or write, at your own risk.

**Comment by robinvanemden** [comment]
> It simply returns a raw pointer to memory. You can read or write, at your own risk.

@soundandform : I created a [pull request](https://github.com/wasm3/wasm3/pull/46) with some minor changes, and tested linear memory access on an `ESP32 Wrover`. Our WASI modules now work beautifully:
```
Loading WebAssembly...
Start of WASI calls...
Hello from WASI!
Raw pointer to start of linear memory: 0x3f800028
Linear memory size in bytes: 196608
Input string length inc zero terminator: 22
Int pointer memory address of WASM mallocated input string: 131296
Input string:{"input":[12,37,100]}
WASM reported string length: 22 | C reported string length: 22
Running add() function.
Int pointer memory address of WASM mallocated output string: 131344
Output string:  {"output": [149]}
```
Thanks again!

**Comment by vshymanskyy** [comment]
Merged #46
